File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:

        def findLast(nums):

            low = 0
            high = len(nums) - 1
            last = -1

            while low <= high:
                mid = (low + high) // 2
                if nums[mid] == target:
                    last = mid
                    low = mid + 1
                elif nums[mid] < target:
                    low = mid + 1
                else:
                    high = mid - 1

            return last

        def findFirst(nums):

            low = 0
            high = len(nums) - 1
            first = -1

            while low <= high:
                mid = (low + high) // 2
                if nums[mid] == target:
                    first = mid
                    high = mid - 1
                elif nums[mid] < target:
                    low = mid + 1
                else:
                    high = mid - 1

            return first

        return [findFirst(nums), findLast(nums)]
        

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java


import javax.xml.crypto.dsig.keyinfo.RetrievalMethod;

class Solution {
    public int findOccur(int[] nums, int target, Boolean leftBias){

        int left = 0;
        int right = nums.length - 1;
        int val = -1;

        while (left <= right) {
            int mid = (left + right) / 2;

            if (nums[mid] > target) {
                right = mid - 1;
            } else if (nums[mid] < target) {
                left = mid + 1; 
            } else {
                if (leftBias) {
                    right = mid - 1;
                    val = mid;
                } else {
                    left = mid + 1;
                    val = mid;
                }
            }
        }
        
        return val;
    }

    public int[] searchRange(int[] nums, int target) {

        int[] result = new int[2];
        result[0] = findOccur(nums, target, true);
        result[1] = findOccur(nums, target, false);

        return result;

    }
}